### PR TITLE
Tests for upload command

### PR DESCRIPTION
**Pull request by ayanginet**
### What was wrong?

Tests for the upload command were missing. Related to #118.

### How was it fixed?

This PR implements these tests.

#### Cute Animal Picture

Cute animal picture

### CODE REVIEW

File: tests/forms/test_address_upload.py
Comment (ayanginet) [inline tests/forms/test_address_upload.py:68]: Hey, @pipermerriam I am not really sure if having all invalid cases under one check is correct. Should I split them, or keep it as it is?

File: tests/forms/test_address_upload.py
Comment (pipermerriam) [inline tests/forms/test_address_upload.py:13]: Generally best practice to avoid single letter variable names, even in tests.

File: tests/forms/test_address_upload.py
Comment (pipermerriam) [inline tests/forms/test_address_upload.py:64]: Here it would be nice to also verify this by actually seeing the records are written to the database

File: tests/forms/test_address_upload.py
Comment (pipermerriam) [reply tests/forms/test_address_upload.py:68]: Generally, when we have independent failure conditions, it is nice to have them tested independently.  Not strictly necessary, but it's a good best practice.

Suppose we have two error conditions A and B.

```python
def test_error_conditions():
    assert not condition_A()
    assert not condition_B()
```

When we write a test this way, then if `A` is failing, we don't actually know whether `B` is also failing since we can't get there until we get `A` passing.  When everything is passing this works fine.  When just one of them is failing it's pretty easy to debug.  However, when there is complicated interplay and maybe the fix for `B` is causing `A` to fail then it gets difficult to debug...

You can typically get some mileage deduplicating tests like this with `@pytest.mark.parametrize`, but it's also ok to be verbose and copy/pasta if the differences in testing logic are subtle enough that parametrization doesn't lead to legible tests.

**Review by pipermerriam — APPROVED** [review]

**Comment by ayanginet** [comment]
Also, I won't be able to merge this because I am not a contributor in this project.

**Comment by ligi** [comment]
I will merge PR's when they are approved by @pipermerriam

**Comment by pipermerriam** [comment]
I'll take a look at this tomorrow or Monday.

Sent from ProtonMail mobile

-------- Original Message --------

File: tests/forms/test_address_upload.py
Comment (pipermerriam) [inline tests/forms/test_address_upload.py:12]: two things.

1. I'm *assuming* that the first argument here is an email address?  I find it nice to use keyword arguments in cases like this to document what the fields are.
2. If the first is an email, I would recommend using `@example.com` email addresses as a best practice.

File: tests/forms/test_address_upload.py
Comment (pipermerriam) [inline tests/forms/test_address_upload.py:11]: The name of this fixture feels disconnected from what it returns.  Maybe call it something like `wallet_address`?

File: tests/forms/test_address_upload.py
Comment (pipermerriam) [inline tests/forms/test_address_upload.py:47]: :heart: love the documentation here.

File: tests/forms/test_address_upload.py
Comment (pipermerriam) [inline tests/forms/test_address_upload.py:98]: I'm not *sure* this is actually wrong as it's been a while since I did a lot of "web" and my standard status code knowledge is rusty but the 2xx codes tend to mean "success".

Some quick googling suggests we should be using something in the 4xx range: https://stackoverflow.com/questions/47269601/what-http-response-code-to-use-for-failed-post-request

I don't think this is something you need to cleanup in this PR (avoid scope creep), but it might be a nice thing to open an issue for?

File: tests/forms/test_address_upload.py
Comment (pipermerriam) [inline tests/forms/test_address_upload.py:152]: *nitpick*

Not necessary, but since this is repeated in each of the tests, you *could* have a fixture like this:

```python
@pytest.fixture
def admin_client(client):
    client.login(email='admin@pretix', password='admin')
    return client
```

But as I'm writing this I realize that is sort of a [foot gun](https://en.wiktionary.org/wiki/footgun) because someone *might* write a test like this:

```python
def test_something(client, admin_client):
    # ...
```

In which case they wouldn't realize that both `client` and `admin_client` are the same object and are both logged in as the admin user...  So maybe go a step further and create a new client instance if you are planning to do this?  I'm not actually sure it's a lot better, but it does remove a  little bit of duplicated code at the added cost of extra fixtures and abstraction layers

**Review by pipermerriam — APPROVED** [review]
Looks solid.  You should have merge permissions now @ayanginet

**Comment by ligi** [comment]
I removed the merge access again. Please let me merge all PRs so I know what is going in the codebase.
@ayanginet I can merge - just give me a signal. Just waiting if you want to address some of the comments of @pipermerriam - I do not think there are blockers in there - still would wait for your ack.
@pipermerriam thanks for the review

File: tests/forms/test_address_upload.py
Comment (ayanginet) [reply tests/forms/test_address_upload.py:84]: Opened new issue #125

**Review by pipermerriam — APPROVED** [review]
@ligi my standard for reviews is:

- comment: just a comment about something and not a full review
- request changes: there is something about the PR that I feel needs to be changed and I would like to review it again once it has been changed.
- approve: Once the author has addressed my comments to **their** satisfaction they may merge at their discretion.

If you (@ligi ) want to be the one who merges, then I think our process here will need to be:

1. ayan/nurtas request review from me
2. I review until PR is satisfactory for me to mark as "approved" 
3. ayan/nurtas signals to @ligi that they have made all the changes they want to make
4. @ligi merges or kicks PR back if there is something else that needs to be addressed.

**Comment by ligi** [comment]
Sounds good to me - looks like we are at step 3 now ;-)

**Comment by ayanginet** [comment]
@ligi I think this PR is ready to be merged 👍

**Pull request merged**
